src/map_memory.cpp

```diff
@@ -0,0 +1,59 @@
+#include "map_memory.h"
+
+#include <algorithm>
+
+void map_memory::trim( int limit )
+{
+    while( tiles.size() > static_cast<size_t>( limit ) ) {
+        tile_map.erase( tiles.back().first );
+        tiles.pop_back();
+    }
+    while( symbols.size() > static_cast<size_t>( limit ) ) {
+        symbol_map.erase( symbols.back().first );
+        symbols.pop_back();
+    }
+}
+
+memorized_terrain_tile map_memory::get_tile( const tripoint &pos ) const
```

Comment:
This can return a constant reference (in case of a static constant for the default value).